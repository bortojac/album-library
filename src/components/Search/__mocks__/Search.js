import React from 'react';
const Search = ({ searchActive }) => <div>A mock with '{searchActive}' passed!</div>;
export default Search;