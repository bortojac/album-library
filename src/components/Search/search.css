.searchContainer {
  display: flex;
  flex-flow: column;
  /*flex: 1;*/
  width: 100%;
  justify-content: center;
  padding-right: 1rem;
  padding-left: 1rem;
  align-items: center;
  overflow: hidden;
  transition: max-height 1s ease;
}

.searchContainer--active {
  max-height: 50%;
}

.searchContainer--closed {
  max-height: 0;
}

.searchContainer__input {
  border: solid .2rem #000;
  margin-top: 2rem;
  font-size: 1.5rem;
  margin-bottom: 1rem;
  padding: .5rem;
  width: 100%;
  box-sizing: border-box;
}

.searchContainer__button {
  width: 5rem;
  padding: .5rem;
  align-self: flex-start;
  margin: 1rem 0;
  border: solid .2rem #000;
  cursor: pointer;
}

@media screen and (min-width: 641px) {

  .searchContainer {
    width: 50%;
  }

}