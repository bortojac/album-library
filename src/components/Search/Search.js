import React, { Component } from 'react';
import AlbumList from '../AlbumList';
import './search.css';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchInput: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  handleChange(e) {
    this.setState({ searchInput: e.target.value });
  }

  // we want enter to trigger search
  handleKeyDown(event) {
    if (event.keyCode === 13 && this.state.searchInput) {
      this.handleSearch();
    }
  }

  handleSearch() {
    if (this.state.searchInput) {
      this.props.fetchSearchResults(this.state.searchInput);
    }
    else {
      return;
    }
  }

  handleClick() {
    this.props.toggleSearch();
  }

  render() {
    return (
      <section
        className={`searchContainer ${this.props.active ? 'searchContainer--active' : 'searchContainer--closed'}`}>
        <input
          className="searchContainer__input"
          placeholder="Search for an Artist or Album"
          onChange={this.handleChange}
          onKeyDown={this.handleKeyDown}
        />
        <AlbumList searchFlag />
        <button className="searchContainer__button" type="button" onClick={this.handleClick}>done</button>
      </section>
    );
  }
}

export default Search;