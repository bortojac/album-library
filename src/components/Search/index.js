import { connect } from 'react-redux';
import { fetchSearchResults } from '../../store/actions';
import Search from './Search';

const mapStateToProps = state => ({ });

const mapDispatchToProps = dispatch => {
  return {
    fetchSearchResults: (term) => dispatch(fetchSearchResults(term)),
  };
};


export default connect(mapStateToProps, mapDispatchToProps)(Search)
