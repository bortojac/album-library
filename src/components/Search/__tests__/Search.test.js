import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Search from '../Search';

jest.mock('../../AlbumList');

// check render
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Search />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot
it('renders correctly', () => {
  const tree = renderer
    .create(<Search />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
