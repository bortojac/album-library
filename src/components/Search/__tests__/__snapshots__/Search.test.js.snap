// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="searchContainer searchContainer--closed"
>
  <input
    className="searchContainer__input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search for an Artist or Album"
  />
  <button
    className="searchContainer__button"
    onClick={[Function]}
    type="button"
  >
    done
  </button>
</section>
`;
